sycl/include/sycl/handler.hpp

```diff
@@ -2865,14 +2865,9 @@ class __SYCL_EXPORT handler {
   /// device copyable.
   /// \param Count is the number of times to fill Pattern into Ptr.
   template <typename T> void fill(void *Ptr, const T &Pattern, size_t Count) {
-    throwIfActionIsCreated();
-    setUserFacingNodeType(ext::oneapi::experimental::node_type::memfill);
     static_assert(is_device_copyable<T>::value,
                   "Pattern must be device copyable");
-    parallel_for<__usmfill<T>>(range<1>(Count), [=](id<1> Index) {
-      T *CastedPtr = static_cast<T *>(Ptr);
-      CastedPtr[Index] = Pattern;
-    });
+    this->fill_impl(Ptr, &Pattern, sizeof(T), Count);
```

Comment:
There's no OpenCL backend that supports SYCL-Graph in DPC++ CI, but I suspect this change will break the Graphs E2E tests that use `handler::fill` for USM, as in cl_khr_command_buffer there is no command that corresponds to a USM fill. See https://github.com/intel/llvm/blob/sycl/sycl/doc/design/CommandGraph.md#limitations

I'll try this branch out locally with an OpenCL cl_khr_command_buffer backend and report back which graphs E2E tests need disabled and if we need a new bullet point for `handler::fill()` in that design doc section I linked.

Reply:
Aha I see, thanks for the help with that! I'll update the tests and the design doc

Reply:
perfect, thanks